### PR TITLE
refactor: refactor component to typescript

### PR DESCRIPTION
**Pull request by jim-deriv**
## Changes:

Please include a summary of the change and which issue is fixed below:
-   ...

## When you need to add unit test

-   [ ] If this change disrupt current flow
-   [ ] If this change is adding new flow

## When you need to add integration test

-   [ ] If components from external libraries are being used to define the flow, e.g. @deriv/components
-   [ ] If it relies on a very specific set of props with no default behavior for the current component.

## Test coverage checklist (for reviewer)

-   [ ] Ensure utility / function has a test case 
-   [ ] Ensure all the tests are passing

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [X] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: packages/components/src/components/progress-slider-mobile/progress-slider-mobile.tsx [hunk 1]
```diff
@@ -1,12 +1,22 @@
...
 import ProgressTicksMobile from './progress-ticks-mobile.jsx';
 import CircularProgress from '../circular-progress';
 import RemainingTime from '../remaining-time';
 import Text from '../text';
 
+type TProgressSliderMobileProps = {
+    className: string;
+    current_tick: number;
+    expiry_time: number & string;
+    is_loading: boolean;
```
Comment (iman-fs) [inline packages/components/src/components/progress-slider-mobile/progress-slider-mobile.tsx:13]: I think this should be optional

File: packages/components/src/components/progress-slider-mobile/progress-slider-mobile.tsx [hunk 2]
```diff
@@ -1,12 +1,22 @@
...
+    server_time: moment.Moment;
+    start_time: number & string;
+    ticks_count: number;
```
Comment (iman-fs) [inline packages/components/src/components/progress-slider-mobile/progress-slider-mobile.tsx:16]: same for this one

**Comment by iman-fs** [comment]
@jim-deriv tests are failing. can you check it please?

**Comment by jim-deriv** [comment]
> @jim-deriv tests are failing. can you check it please?

@iman-fs, This card is in blocked, why are you checking it? It needs changes from one of the cards. It's not yet in announce_fe_review

**Pull request closed without merging**